lilypad/developer-resources/create-lilypad-module/building-your-module.md

```diff
@@ -0,0 +1,349 @@
+
+For now, let's look at the top 2 lines of the provided code block:
+
+```python
+import torch
+from transformers import DistilBertTokenizer, DistilBertForSequenceClassification
+```
+
+Notice that `torch` is also being used. Copy the `transformers` import statement and paste it over the existing import statement in our `download_models.py` file.
+
+Now open `requirements.txt`:
+
+```txt
+# torch==2.4.1
+# transformers==4.47.1
+```
+
+These are 2 of the most common libraries when working with models. Similar to the `import` statement in the `download_models.py` file, they are provided by default for convenience, but commented out because although they are common, not every model will use them.
+
+Since this model happens to use both of these libraries, we can uncomment both lines and close the file after saving.
+
+{% hint style="info" %}
+`torch` is a collection of APIs for extending [PyTorch](https://pytorch.org/)’s core library of operators.
+
+- [Learn more about using `datasets` with `Pytorch`](https://huggingface.co/docs/datasets/en/use_with_pytorch)
+- [Read the `torch` documentation](https://pytorch.org/docs/stable/index.html)
+  {% endhint %}
+
+Return to the `download_models.py` file, and look for the next `TODO` comment.
+
+```python
+# TODO: Set this to your model's Hugging Face identifier
+MODEL_IDENTIFIER = ""
+```
+
+If we take a look at the [Distilbert Hugging Face page](https://huggingface.co/distilbert/distilbert-base-uncased-finetuned-sst-2-english), we can use the copy button next to the name of the module to get the `MODULE_IDENTIFIER`. Just paste that in as the value.
+
+For our use case, it should look like this:
+
+```python
+MODEL_IDENTIFIER = "distilbert/distilbert-base-uncased-finetuned-sst-2-english"
+```
+
+You're almost ready to download the model. All you need to do now is replace the following 2 lines:
+
+```python
+tokenizer = AutoTokenizer.from_pretrained(MODEL_IDENTIFIER)
+model = AutoModelForSequenceClassification.from_pretrained(MODEL_IDENTIFIER)
+```
+
+Instead of using `AutoTokenizer` and `AutoModelForSequenceClassification`, replace those with the `DistilBertTokenizer` and `DistilBertForSequenceClassification` we imported.
+
+```python
+tokenizer = DistilBertTokenizer.from_pretrained(MODEL_IDENTIFIER)
+model = DistilBertForSequenceClassification.from_pretrained(MODEL_IDENTIFIER)
+```
+
+The script is now configured! Try running the command again.
+
+```sh
+python -m scripts.download_models
+```
+
+The `models` directory should now appear in your project. 🎉
+
+{% hint style="info" %}
+No matter which model you are using, be sure to thoroughly read the model's documentation to learn how to properly download and use the model locally.
+{% endhint %}
+
+## Preparing Your Module
+
+Before you are able to run your module, we need to build the Docker image. You can run the following command:
```

Comment:
Im confused here. Before you are able to run your module, the inference script should be created?

Reply:
the idea was that we prepare the module to be built (although the module isn't implemented yet) so that all the configuration is dealt with first and then we can focus on building the module. it was a last minute decision to move this section before the "Building Your Module" section and am definitely open to moving it back if you think that flows better.

Reply:
My concern here is that each build that includes code changes will create a new image. This means that when a user runs the module locally, they will end up with one working image while the previous one remains unused, potentially taking up disk space.

I think the most optimistic approach would be to only have the user build the image once they have the code ready to run.

Reply:
that is a good point, tbh i thought when building a Docker image with the same tag and name as another Docker image the old one would get removed, but i just tried it and it just becomes untagged.

on another note, should we include some basic instructions on how to clean these images up? or are we already doing enough by providing these scripts to begin with? there is a note recommending people to get familiar with Docker already